rfc/rfc023-x509credential.md

```diff
@@ -0,0 +1,556 @@
+* Subject Other Name (san)
+  * email
+  * dns
+  * uri
+* Extended Key Usage (eku)
+  * Any OID
+* A Free-to-Use CA For Code Signing (fulcio-issuer)
+  * Any issuer hostname
+
+### Extending the `did:x509` specification
+
+This RFC extends the Subject Other Name (san) policy with the following attribute:
+
+* Subject Other Name (san)
+  * otherName: A free-form attribute that can be used to specify any attribute that is not covered by the other
+    attributes.
+
+The otherName attribute can be used to specify extra attributes in a x509 certificate. This attribute is added to the specification of this RFC to cater for the use case where the san.otherName attribute is used in the x509 certificate and plays a role in the identification of the holder of the certificate.
+
+## The NutsX509Credential Structure
+
+An `NutsX509Credential` must conform to the general structure of a W3C Verifiable Credential and conform to the following
+rules:
+
+- The credential MUST be in JWT format.
+- `type`: MUST include `VerifiableCredential` and `NutsX509Credential`.
+- `issuer`: MUST be a valid `did:x509` identifier.
+- `credentialSubject`: MUST only contain fields explicitly present in the `did:x509` DID policies with the fields mapped by each type as a separate map. An example: 
+```
+{
+  "subject": {
+    "O" : "My Organisation"
+  },
+  "san" : {
+    "email" : "info@example.com"
+  }
+}
+```
+
+The credential subject can be identified by any DID method (e.g. `did:web`) accepted by the credential verifier.
+
+### Example `NutsX509Credential`
+
+Below is an example of an `NutsX509Credential` issued by a `did:x509` DID. The credential subject is identified by a
+`did:web`.
+The first snippet is the JWT header, and the second snippet is the credential payload.
+
+(TODO: check this example)
+
+```json
+{
+  "alg": "PS256",
+  "typ": "JWT",
+  "x5c": [
+    "<base64 encoded leaf-certificate in the DER format>",
+    "<base64 encoded issuer-intermediate-certificate in the DER format>",
+    "<base64 encoded issuer-certificate in the DER format>"
+  ],
+  "x5t": "<thumbprint>",
+  "kid": "did:x509:0:sha256:WE4P5dd8DnLHSkyHaIjhp4udlkF9LqoKwCvu9gl38jk::subject:O:OLVG%20Oost::subject:L:Amsterdam::san:otherName:23419943234#1"
+}
+```
+
+(TODO: Add right JSON-LD context)
+Payload:
+
+```json
+{
+  "vc": {
+    "@context": [
+      "https://www.w3.org/2018/credentials/v1"
+    ],
+    "type": [
+      "VerifiableCredential",
+      "NutsX509Credential"
+    ],
+    "issuer": "did:x509:0:sha256:WE4P5dd8DnLHSkyHaIjhp4udlkF9LqoKwCvu9gl38jk::subject:O:OLVG%20Oost::subject:L:Amsterdam::san:otherName:23419943234",
+    "issuanceDate": "2024-12-01T00:00:00Z",
+    "credentialSubject": {
+      "id": "did:web:example.com",
+      "subject:O": "OLVG Oost",
+      "subject:L": "Amsterdam",
+      "san:otherName": "23419943234"
+    }
+  }
+}
+```
+
+## Validation
+
+### 1. Verify Credential Structure
+
+To validate an `NutsX509Credential`, the following steps MUST be performed:
```

Comment:
It's just `X509Credential` (without the Nuts prefix)

Reply:
I'm confused now, nuts node:
```
	X509CredentialType = "X509Credential"
```
go-didx509-toolkit:
```
var CredentialType = ssi.MustParseURI("X509Credential")
```

Reply:
Renaming it.